Improve key concepts page (#1020)
wso2/docs-mi: en/docs/get-started/key-concepts.md

See the [Sequences]({{base_path}}/reference/mediation-sequences) documentation for more information.

#### Message Stores and Processors

A Message Store is used to temporarily store messages before they are delivered to their destination. This approach is useful for several scenarios.

1. Serving traffic to back-end services that can only accept messages at a given rate, whereas incoming traffic arrives at different rates. This use case is called request rate matching.
2. If the back-end service is not available at a particular moment, the message can be kept safely inside the message store until the back-end service becomes available. This use case is called guaranteed delivery. 

The task of the Message Processor is to pick the messages stored in the message store and deliver them to the destination.

See the [Message Stores and Processors]({{base_path}}/reference/synapse-properties/about-message-stores-processors) documentation for more information.

#### Templates

> A Template is a reusable configuration that help streamline and simplify the creation and management of [sequences](#sequences) and [endpoints](#endpoints). Templates allow developers to abstract common logic and reuse it across multiple services, reducing redundancy and ensuring consistency in configurations.

Reviewer: ```suggestion
A Template is a reusable configuration that helps streamline and simplify the creation and management of [sequences](#sequences) and [endpoints](#endpoints). Templates allow developers to abstract common logic and reuse it across multiple services, reducing redundancy and ensuring consistency in configurations.
```

See the [Templates]({{base_path}}/reference/synapse-properties/template-properties) documentation for more information.

---

### Message exit points

#### Endpoints

> An Endpoint defines an external destination for a message. An endpoint could represent an HTTP URL, a mailbox, a JMS queue, a TCP socket, etc., along with the settings needed for the connection. 

Reviewer: ```suggestion
An endpoint defines an external destination for a message. It could represent an HTTP URL, a mailbox, a JMS queue, a TCP socket, and other connection types, along with the settings needed for the connection.
```

See the [Endpoints]({{base_path}}/reference/synapse-properties/endpoint-properties) documentation for more information.

#### Connectors

A Connector allows your mediation flows to connect and interact with external services such as Twitter and Salesforce. Typically, connectors are used to wrap the API or the SDK of an external service. Each connector provides operations that perform different actions in that service. For example, the Twitter connector has operations for creating a tweet, getting a user's followers, and more.

To download a required connector, go to the [WSO2 Connector Store](https://store.wso2.com/store).

See the [Connectors]({{base_path}}/reference/connectors/connectors-overview) documentation for more information.

---

### Other concepts

#### Data Services

> A Data service allow users to expose data from various data sources such as databases, spreadsheets, and CSV files through services, enabling easy access to this data over HTTP. WSO2 MI simplifies the process of interacting with databases by abstracting the data access logic and offering a way to expose data as services, such as RESTful or SOAP APIs, without requiring developers to write complex code.

Reviewer: ```suggestion
A Data service allows users to expose data from various data sources such as databases, spreadsheets, and CSV files through services, enabling easy access to this data over HTTP. WSO2 MI simplifies the process of interacting with databases by abstracting the data access logic and offering a way to expose data as services, such as RESTful or SOAP APIs, without requiring developers to write complex code.
```

See the [Data Services]({{base_path}}/reference/synapse-properties/data-services) documentation for more information.

#### Scheduled Tasks

Executing an integration process at a specified time is a common requirement in enterprise integration. For example, in an organization, there can be a need to run an integration process to synchronize two systems every day at the end of the day. In the Micro Integrator, the execution of a message mediation process can be automated to run periodically by using a scheduled task. Furthermore, you can use cron expressions for more advanced scheduling configurations.

See the [Scheduled Tasks]({{base_path}}/reference/synapse-properties/scheduled-task-properties) documentation for more information.

#### Transports

> A transport protocol is responsible for carrying messages that are in a specific protocols. WSO2 Micro Integrator supports all the widely used transports including, HTTP/S, JMS, VFS, as well as domain-specific transports like FIX. Each transport provides a receiver implementation for receiving messages and a sender implementation for sending messages.

Reviewer: ```suggestion
A transport protocol is responsible for carrying messages that are in a specific protocol. WSO2 Micro Integrator supports all the widely used transports including, HTTP/S, JMS, and VFS, as well as domain-specific transports like FIX. Each transport provides a receiver implementation for receiving messages and a sender implementation for sending messages.
```

See the [Transports]({{base_path}}/install-and-setup/setup/transport-configurations/configuring-transports) documentation for more information.

#### Registry

WSO2 Micro Integrator uses a Registry to store various configurations and resources, such as [endpoints](#endpoints). A registry is simply a content store and a metadata repository. Various resources such as XSLT scripts, WSDLs, and configuration files can be stored in a registry and referred to by a key, which is a path similar to a UNIX file path. The WSO2 Micro Integrator uses a [file-based registry]({{base_path}}/install-and-setup/setup/deployment/file-based-registry) that is configured by default. You can also define and use a [local registry]({{base_path}}/develop/creating-artifacts/registry/creating-local-registry-entries) when you develop your integration artifacts,

#### Message Builders and Formatters

> When a message comes into WSO2 Micro Integrator, the receiving transport selects a message builder based on the message's content type. It uses that builder to process the message's raw payload data and converts it to 

Reviewer: ```suggestion
It uses that builder to process the raw payload data of the message and converts it to a
```

See the [Proxy Services]({{base_path}}/reference/synapse-properties/proxy-service-properties) documentation for more information.

#### Inbound Endpoints

In [APIs](#apis) and [proxy services](#proxy-services) some parts of the configuration are global to a particular instance. For example, the HTTP port needs to be common for all the APIs. 
The Inbound Endpoints do not contain such global configurations. That gives extra flexibility in configuring 
the Inbound Endpoints compared to the other two message entry points.

See the [Inbound Endpoints]({{base_path}}/reference/synapse-properties/inbound-endpoints/about-inbound-endpoints) documentation for more information.

---

### Message processing units

#### Mediators

Mediators are individual processing units that perform a specific function on messages that pass through the Micro Integrator. 
The mediator takes the message received by the [message entry points](#message-entry-points), 
carries out some predefined actions on it (such as transforming, enriching, filtering), and modifies the message.

See the [Mediators]({{base_path}}/reference/mediators/about-mediators) documentation for more information.

#### Sequences

> A sequence is a set of [mediators](#mediators) organized into a logical flow, allowing you to implement [pipes and filter patterns]({{base_path}}/learn/enterprise-integration-patterns/messaging-systems/pipes-and-filters). The mediators in the sequence will perform the necessary message processing and route the message 

Reviewer: Shall we remove the extra line break here?
